src/inspect.rs

```diff
@@ -247,10 +254,75 @@ pub fn handle_inspect_snapshot(inspect_snapshot: &InspectSnapshot) -> Result<()>
         BiscuitBytes::FromFile(snapshot_format, inspect_snapshot.snapshot_file.clone())
     };
 
+    let authorizer_from = match (
+        &inspect_snapshot.authorization_args.authorize_interactive,
+        &inspect_snapshot.authorization_args.authorize_with,
+        &inspect_snapshot.authorization_args.authorize_with_file,
+    ) {
+        (false, None, None) => None,
+        (true, None, None) => Some(DatalogInput::FromEditor),
+        (false, Some(str), None) => Some(DatalogInput::DatalogString(str.to_owned())),
+        (false, None, Some(path)) => Some(DatalogInput::FromFile(path.to_path_buf())),
+        // the other combinations are prevented by clap
+        _ => unreachable!(),
+    };
+
+    if let Some(vf) = &authorizer_from {
+        ensure_no_input_conflict(vf, &snapshot_from)?;
+    }
+
```

Comment:
There is a lot of common behavior between `handle_inspect` and `handle_inspect_snapshot`. Factorize in functions avoid code dedup and ease read.

Reply:
yeah, the issue is that both inspect and inspect snapshot interleave IO and computation. The JSON PR already improves on that, but that will require dedicated work.